### DISCUSSION TITLE
How do we use the tax lot and zoning district tilesets with Deck's MVTLayer?

### DISCUSSION THREAD

**Discussion by TangoYankee** [NYCPlanning/ae-zoning-map-poc#32] [Q&A]
# Description

Zoning districts and tax lots both have the same question. But for simplicity, I will focus on zoning districts (zd).

The zd tilesets have the following structure:

![zoning-disctrict tileset](https://github.com/NYCPlanning/ae-zoning-map-poc/assets/11068062/32e0a2fe-5676-4633-98ab-b32f350ee9be)

When accessing the tileset through an MVT layer, two key pieces of data are rendered under the domain of this single layer.
1) Both fill and label features are rendered together
    - This complicates the setting of the getFillColor; labels and fill need different color logic but they are controlled by the same   function
    - This concern is alleviated by the presence of a 'layerName' property that distinguishes the two and can be used for conditional statements to change the color logic
2) All zd categories are rendered together. 
   - This complicates the setting of visibility; we want to make each category individually visible. However, control is only at the layer level
   - It also complicates the 'blending' of colors when a zoning district is part of two categories- we would no longer be able to render two colors over each other.   

```  
const zoningDistrictsLayer = new MVTLayer({
    id: "zoningDistricts",
    data: `${STORAGE_URL}/tilesets/zoning_district/{z}/{x}/{y}.pbf`,
    visible: boolean,
    getFillColor: (f) =>   rgba});
```

Is there a way to take the data of the tileset and separate it into different layers? As in, could we modify the `id` or `data` parameter to only get the layer fill or label data, then creating multiple layers with different data?

Would it make sense to use multiple layers with the same data but with different logic? For instance, a 'commercial fill' layer would only be visible when 'commercial' is toggled on. It's getFillColor function would hide the other data elements by turning the opacity of the other categories down to zero. This could bring back color blending

Would it be easiest to stick with one layer but have intricate 'if' statements that style each type of data differently?

Should we just split the data into more tilesets?

@TylerMatteo, do you have thoughts?
@dhochbaum-dcp and @bmarchena, have you started experimenting with this?

**Comment by dhochbaum-dcp** [comment]
I've taken a look at it and my suggestion is to update the opacity to show and hide things that we cannot toggle using visible.

**Comment by TangoYankee** [comment]
Relevant comment in discussion #26 https://github.com/NYCPlanning/ae-zoning-map-poc/discussions/26#discussioncomment-7766572
